### PR TITLE
Update API-design-guidelines.md-with-wildcard-scopes-icm-agreement

### LINKED ISSUE #184: Clear documentation on the use of wildcard scope in Camara
**Problem description**
The current API design guidelines in section [11.6](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#116-security-definition) points to CAMARA-API-access-and-user-consent.md doc in ICM WG to follow security guidelines and then goes in detail to describe scope-naming in section [11.6.1](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#1161-scope-naming). 

We do not explicitly document in design guidelines if Camara supports the use of wildcard scope (to represent "all scopes included case") in addition to the usual granular scopes. This is only documented in the [applying-purpose-concept-section](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#applying-purpose-concept-in-the-authorization-request) of the CAMARA-API-access-and-user-consent.md document. 

We need to clearly document if we support the use of wildcard scope in Camara and add this in the design guidelines and if there is a decision to support wildcard scope, the naming convention for such a wildcard scope has to be documented in design guidelines. Every subproject needs to also add this within the spec file so that we do not have a situation where some providers support it and some not.

**Expected action**
Get a consensus in commonalities on the support of wildcard scopes in Camara and if we agree to support this, it should be clearly documented in design guidelines and the individual API spec files.

**Additional context**
- This also impacts https://github.com/camaraproject/Commonalities/issues/163 

- The current [PR](https://github.com/camaraproject/IdentityAndConsentManagement/pull/121) in ICM which deals with the Camara OIDC Profile mentions the scope parameter and says that it should follow the API spec file listed scopes and this would mean that if we support wildcard scope, it has to make its way into the spec file https://github.com/camaraproject/IdentityAndConsentManagement/blob/a97ffae7ffa091796bbde424428b660fe7418330/documentation/CAMARA-Security-Interoperability.md#the-scope-parameter

### PR DESCRIPTION
**Pull request by shilpa-padgaonkar**
#### What type of PR is this?

Add one of the following kinds:
* documentation

#### What this PR does / why we need it:
This PR includes the following changes:

- Documents the agreement in ICM on wildcard scopes discussed under the issue -(https://github.com/camaraproject/IdentityAndConsentManagement/issues/95) into the design guidelines to ensure completeness of the doc,

- The explicit subscriptions related scope changes were documented in the PR https://github.com/camaraproject/Commonalities/pull/177 .  The scope naming is described in 2 sections in the design doc - 10.2 and 11.6.1. The earlier PR only updated section 10.2 (b). This PR adds the needed content also in section 11.6.1

#### Which issue(s) this PR fixes:

Fixes #184 

#### Changelog input

```
 release-note

```

#### Additional documentation 

This section can be blank.

```
docs

```

### CODE REVIEW

**Comment by shilpa-padgaonkar** [comment]
@AxelNennker and @jpengar Could you kindly check if the section "API-level scopes" reflects correctly the agreement made in https://github.com/camaraproject/IdentityAndConsentManagement/issues/95

@PedroDiez and @bigludo7 : I have simply duplicated the content of https://github.com/camaraproject/Commonalities/pull/177 into the section 11.6.1 to ensure consistency. Could you kindly take a look?

File: documentation/API-design-guidelines.md [hunk 1]
```diff
@@ -1297,17 +1297,37 @@ where
...
+For e.g. device-roaming-subscriptions:read
+
+The format to define scopes for explicit subscriptions with action create, includes the event type in its formulation to ensure that consent is managed at the level of subscribed event types. Scopes should be represented as below for APIs that offer explicit event subscriptions with action create:
+
+API Name: device-roaming-subscriptions
+Event-type: org.camaraproject.device-roaming-subscriptions.v0.roaming-on
+Grant-level, action on resource: create
+For e.g. device-roaming-subscriptions:org.camaraproject.device-roaming-subscriptions.v0.roaming-on:create
+
+##### API-level scopes (sometimes referred to as wildcard scopes in Camara)
```
Comment (jpengar) [inline documentation/API-design-guidelines.md:1328]:
```suggestion
##### API-level scopes (sometimes referred to as wildcard scopes in CAMARA)
```

Comment (jpengar) [reply documentation/API-design-guidelines.md:1328]: I'm not sure even if just putting `##### API-level scopes` would be ok. Not to mention the wildcard thing...

**Review by jpengar — APPROVED** [review]
LGTM. I've just added some minor comment.

File: documentation/API-design-guidelines.md [hunk 2]
```diff
@@ -1297,17 +1297,37 @@ where
...
+The decision on the API-level scopes was made within the ICM working group and is documented in the design guidelines to ensure completeness of this document. The scopes will always be those defined in the API Specs YAML files. Thus, a scope would only provide access to all endpoints and resources of an API if it is explicitly defined in the API Spec YAML file and agreed in the corresponding API subproject.
```
Comment (jpengar) [inline documentation/API-design-guidelines.md:1329]:
```suggestion
The decision on the API-level scopes was made within the [ICM working group](https://github.com/camaraproject/IdentityAndConsentManagement) and is documented in the design guidelines to ensure completeness of this document. The scopes will always be those defined in the API Specs YAML files. Thus, a scope would only provide access to all endpoints and resources of an API if it is explicitly defined in the API Spec YAML file and agreed in the corresponding API subproject. 
```

**Review by jpengar — APPROVED** [review]

File: documentation/API-design-guidelines.md [hunk 3]
```diff
@@ -1297,17 +1297,37 @@ where
...
 | API | path | method | scope |
 | --- | --- | --- | --- |
 | location-verification | /verify | POST | `location-verification:verify` |
 | qod | /sessions | POST | `qod:sessions:create`, or<br>`qod:sessions:write` |
 | qod | /qos-profiles | GET | `qod:qos-profiles:read` |
 
+<br>
+
+
+##### Regarding scope naming for APIs which deal with explicit subscriptions, the guidelines propose some changes as compared to the above format and this is described below:
```
Comment (rartych) [inline documentation/API-design-guidelines.md:1313]: I think the header level should be the same as in line 1279
```suggestion
#### Regarding scope naming for APIs which deal with explicit subscriptions, the guidelines propose some changes as compared to the above format and this is described below:
```

File: documentation/API-design-guidelines.md [hunk 4]
```diff
@@ -1297,17 +1297,37 @@ where
...
+For e.g. device-roaming-subscriptions:read
+
+The format to define scopes for explicit subscriptions with action create, includes the event type in its formulation to ensure that consent is managed at the level of subscribed event types. Scopes should be represented as below for APIs that offer explicit event subscriptions with action create:
+
+API Name: device-roaming-subscriptions
+Event-type: org.camaraproject.device-roaming-subscriptions.v0.roaming-on
+Grant-level, action on resource: create
+For e.g. device-roaming-subscriptions:org.camaraproject.device-roaming-subscriptions.v0.roaming-on:create
+
+##### API-level scopes (sometimes referred to as wildcard scopes in CAMARA)
```
Comment (rartych) [inline documentation/API-design-guidelines.md:1328]: I think the header level should be the same as in line 1279
```suggestion
#### API-level scopes (sometimes referred to as wildcard scopes in CAMARA)
```

File: documentation/API-design-guidelines.md [hunk 5]
```diff
@@ -1297,17 +1297,37 @@ where
...
+The decision on the API-level scopes was made within the [ICM working group](https://github.com/camaraproject/IdentityAndConsentManagement) and is documented in the design guidelines to ensure completeness of this document. The scopes will always be those defined in the API Specs YAML files. Thus, a scope would only provide access to all endpoints and resources of an API if it is explicitly defined in the API Spec YAML file and agreed in the corresponding API subproject.
```
Comment (jpengar) [inline documentation/API-design-guidelines.md:1329]:
```suggestion
The decision on the API-level scopes was made within the [Identity and Consent Management Working Group](https://github.com/camaraproject/IdentityAndConsentManagement) and is documented in the design guidelines to ensure completeness of this document. The scopes will always be those defined in the API Specs YAML files. Thus, a scope would only provide access to all endpoints and resources of an API if it is explicitly defined in the API Spec YAML file and agreed in the corresponding API subproject. 
```
To be consistent with section 11.6

**Review by bigludo7 — APPROVED** [review]
/LGFM pending the resolution of @rartych and @jpengar comments

**Review by rartych — DISMISSED** [review]
LGTM

**Review by rartych — APPROVED** [review]

**Review by jpengar — APPROVED** [review]
LGTM

**Review by AxelNennker — APPROVED** [review]

**Review by PedroDiez — APPROVED** [review]
LGTM

**Pull request merged**